main.go

```diff
@@ -0,0 +1,605 @@
+	Nonce                *hexutil.Uint64 `json:"nonce"`
+
+	// We accept "data" and "input" for backwards-compatibility reasons.
+	// "input" is the newer name and should be preferred by clients.
+	// Issue detail: https://github.com/ethereum/go-ethereum/issues/15628
+	Data  *hexutil.Bytes `json:"data"`
+	Input *hexutil.Bytes `json:"input"`
+
+	// Introduced by AccessListTxType transaction.
+	AccessList *types.AccessList `json:"accessList,omitempty"`
+	ChainID    *hexutil.Big      `json:"chainId,omitempty"`
+}
+
+type BlockChainAPI struct {
+	flowClient *grpc.Client
+}
+
+// eth_chainId
+// ChainId is the EIP-155 replay-protection chain id for the current Ethereum chain config.
+//
+// Note, this method does not conform to EIP-695 because the configured chain ID is always
+// returned, regardless of the current head block. We used to return an error when the chain
+// wasn't synced up to a block where EIP-155 is enabled, but this behavior caused issues
+// in CL clients.
+func (api *BlockChainAPI) ChainId() *hexutil.Big {
+	return (*hexutil.Big)(big.NewInt(777))
+}
+
+// eth_blockNumber
+// BlockNumber returns the block number of the chain head.
+func (api *BlockChainAPI) BlockNumber() hexutil.Uint64 {
+	return hexutil.Uint64(65848272)
+}
+
+// eth_syncing
+// Syncing returns false in case the node is currently not syncing with the network. It can be up-to-date or has not
+// yet received the latest block headers from its pears. In case it is synchronizing:
+// - startingBlock: block number this node started to synchronize from
+// - currentBlock:  block number this node is currently importing
+// - highestBlock:  block number of the highest block header this node has received from peers
+// - pulledStates:  number of state entries processed until now
+// - knownStates:   number of known state entries that still need to be pulled
+func (api *BlockChainAPI) Syncing() (interface{}, error) {
+	return false, nil
+}
+
+// eth_sendRawTransaction
+// SendRawTransaction will add the signed transaction to the transaction pool.
+// The sender is responsible for signing the transaction and using the correct nonce.
+func (api *BlockChainAPI) SendRawTransaction(
+	ctx context.Context,
+	input []byte,
+) (common.Hash, error) {
+	return common.Hash{}, nil
+}
+
+// accessListResult returns an optional accesslist
+// It's the result of the `debug_createAccessList` RPC call.
+// It contains an error if the transaction itself failed.
+type accessListResult struct {
+	Accesslist *types.AccessList `json:"accessList"`
+	Error      string            `json:"error,omitempty"`
+	GasUsed    hexutil.Uint64    `json:"gasUsed"`
+}
+
+// eth_createAccessList
+// CreateAccessList creates an EIP-2930 type AccessList for the given transaction.
+// Reexec and BlockNrOrHash can be specified to create the accessList on top of a certain state.
+func (s *BlockChainAPI) CreateAccessList(
+	ctx context.Context,
+	args TransactionArgs,
+	blockNrOrHash *rpc.BlockNumberOrHash,
```

Comment:
nit:
```suggestion
	BlockNumberOrHash *rpc.BlockNumberOrHash,
```

Reply:
Updated in https://github.com/onflow/flow-evm-gateway/pull/10/commits/62a9b84faaca485389bd63c4d8a97c12f1c825fb